### DISCUSSION TITLE
Mylący komunikat podczas wystawiania ofert, niedozwolony parametr zależny po wybraniu danej wartości parametru

### DISCUSSION THREAD

**Discussion by hahyes** [allegro/allegro-api#6691] [Suggestions - rejected]
Dzień dobry,

podczas wystawiania, gdy użyje się parametru niedozwolonego na bazie wymagań z API oraz wybranych wartości parametrów, wychodzi błąd o takiej treści:

```
Parametr zależny Odcień nie może przyjąć wartości [fuksja], gdy parametr Kolor ma wartości [różowy].
```

To jest bardzo mylący komunikat. Sugeruje on, że dla koloru różowego odcień przyjmuje jakiś inny nieokreślony zbiór wartości. Nie jest to prawda, ponieważ Odcień może być wypełniany wyłącznie po wybraniu odpowiednich wartości parametru Kolor.

Sugeruję zmianę treści komunikatu na bardziej sensowną, np.

```
Parametr zależny Odcień nie może zostać wypełniony, gdy parametr Kolor ma wartości [różowy].
```

W takim wypadku sprawa byłaby jasna.

**Comment by PrzemyslawLukanowski** [comment]
Dziękuję za sugestię. Czy możesz podać jeszcze przykładowy numer kategorii, gdzie otrzymałeś błąd tego rodzaju?

**Reply by hahyes** [reply]
@PrzemyslawLukanowski Tak, Odzież damska > Sukienki (id: 124264)

**Reply by PrzemyslawLukanowski** [reply]
Dziękuję. W tym przypadku komunikat jest prawidłowy- w parametrze "Odcień", w polu "options.displayDependsOnValueIds", znajdziesz zbiór wartości parametru "nadrzędnego", od których zależy, czy wyświetlamy dany parametr. Znajduje się wśród nich wartość "różowy" (249512_1647415) z parametru "Kolor". W parametrze "Odcień", w polu "dependsOnValueIds", widnieje informacja, jaką wartość parametru "Kolor" musisz wybrać, aby skorzystać z danej wartości dla parametru "Odcień". Przekazałeś wartość "fuksja", którą można wybrać tylko, jeśli w "Kolor" wybrano "249512_1647423" (fioletowy) lub "249512_1647416" (czerwony). 
Zaproponowany przez Ciebie komunikat nie byłby do końca poprawny, ponieważ "Odcień" może zostać wypełniony, jeśli "Kolor" jest "różowy" - przykładowe poprawne wartości: 

```
                    "id": "249513_1647438",
                    "value": "arbuzowy",
                    "dependsOnValueIds": [
                        "249512_1647415"
                    ]
                },
```

```
                {
                    "id": "249513_1647439",
                    "value": "biskupi",
                    "dependsOnValueIds": [
                        "249512_1647415",
                        "249512_1647423"
                    ]
                },
```

**Reply by hahyes** [reply]
@PrzemyslawLukanowski Przepraszam, źle to zinterpretowałem. Zmiana komunikatu w takim razie nie ma sensu.

Mam jeszcze jedno pytanie: czy może wystąpić sytuacja, kiedy `displayDependsOnValueIds` wskazuje na jakieś wartości i zarazem dla żadnej wartości słownika nie ma wypełnionego `dependsOnValueIds`? Nie udało mi się znaleźć takiej sytuacji, ale pomogłoby mi to dobrze zakodować tą funkcjonalność w moim formularzu wystawiania w aplikacji.

**Reply by PrzemyslawLukanowski** [reply]
Nie, taka sytuacja nie powinna wystąpić.
